Automattic/wp-calypso
# Offer Reset: always displays owned products regardless of the selected term

#### Changes proposed in this Pull Request

* Follow the pattern already established in the previous version of the Plans grid. That pattern says that the owned plan and products should always be visible, and highlighted as such, regardless of the selected term. This means that, for example, users should not be able to purchase Jetpack Backup Daily monthly when they already own the yearly version. I our current state, they can.
* Offer a path to upgrade a monthly plan subscription to a yearly one.

#### Testing instructions

Try to pay attention to duplicate cards, label of call to actions, and badges. Everything should make sense. When testing, make sure to purchase both monthly and yearly subscriptions.

There are several cases we should cover to make sure we got all right.

* Individual purchases:
  * Plans: purchase a plan and make sure the card of the plan is carried over regardless the selected term. Make sure products included in the plan behave the same.
  * Products with option: purchase a product, such as Jetpack Backup Daily, and make sure that the card of the product is carried over regardless the selected term.
  * Products without option: purchase a product, such as Jetpack Search, and make sure that the card of the product is carried over regardless the selected term.

* Multiple purchases:
   * Plan + product: purchase a plan and a product. An example could be Jetpack Security Real-time and Jetpack Search. Verify that the cards of the plan and product are carried over regardless the selected term.

* Upgrade to yearly:
   * Purchase a monthly subscription to any plan and selected the Yearly term from the filter bar.
   * Verify that the CTA label says: "Upgrade to Yearly".

Fixes 1169247016322522-as-1192817246446001

#### Demo

##### Jetpack Backup Daily
###### Monthly
![image](https://user-images.githubusercontent.com/3418513/92634732-1e4e5180-f2ab-11ea-91cb-79722e5cb068.png)
###### Yearly
![image](https://user-images.githubusercontent.com/3418513/92634740-21e1d880-f2ab-11ea-9589-7d84dc67cfe5.png)

Jetpack Security
###### Monthly
![image](https://user-images.githubusercontent.com/3418513/92634566-de876a00-f2aa-11ea-9437-d282d1b18477.png)
###### Yearly (Upgrade to Yearly)
![image](https://user-images.githubusercontent.com/3418513/92634581-e34c1e00-f2aa-11ea-8998-b381dd368513.png)
